go/vt/vtgate/planbuilder/testdata/filter_cases.json

```diff
@@ -4920,5 +4965,149 @@
+      "Instructions": {
+        "OperatorType": "Route",
+        "Variant": "Scatter",
+        "Keyspace": {
+          "Name": "user",
+          "Sharded": true
+        },
+        "FieldQuery": "select id, col1 from unq_binary_idx where 1 != 1",
+        "Query": "select id, col1 from unq_binary_idx where col1 between 10 and 50",
+        "Table": "unq_binary_idx"
+      },
+      "TablesUsed": [
+        "user.unq_binary_idx"
+      ]
+    }
+},
+{
+  "comment": "Between clause on multicolumn vindex (cola,colb)",
+  "query": "select cola,colb,colc from multicol_tbl where cola between 1 and 5",
+  "plan": {
+    "QueryType": "SELECT",
+    "Original": "select cola,colb,colc from multicol_tbl where cola between 1 and 5",
+    "Instructions": {
+      "OperatorType": "Route",
+      "Variant": "Scatter",
+      "Keyspace": {
+        "Name": "user",
+        "Sharded": true
+      },
+      "FieldQuery": "select cola, colb, colc from multicol_tbl where 1 != 1",
+      "Query": "select cola, colb, colc from multicol_tbl where cola between 1 and 5",
+      "Table": "multicol_tbl"
+    },
+    "TablesUsed": [
+      "user.multicol_tbl"
+    ]
   }
+},
+{
+  "comment": "Between clause on a binary vindex field with values from a different table",
+  "query": "select s.oid,s.col1, se.colb from sales s join sales_extra se on s.col1 = se.cola where s.oid between se.start and se.end",
+  "plan": {
+    "QueryType": "SELECT",
+    "Original": "select s.oid,s.col1, se.colb from sales s join sales_extra se on s.col1 = se.cola where s.oid between se.start and se.end",
+    "Instructions": {
+      "OperatorType": "Join",
+      "Variant": "Join",
+      "JoinColumnIndexes": "R:0,R:1,L:0",
+      "JoinVars": {
+        "se_cola": 1,
+        "se_end": 3,
+        "se_start": 2
+      },
+      "TableName": "sales_extra_sales",
+      "Inputs": [
+        {
+          "OperatorType": "Route",
+          "Variant": "Scatter",
+          "Keyspace": {
+            "Name": "user",
+            "Sharded": true
+          },
+          "FieldQuery": "select se.colb, se.cola, se.`start`, se.`end` from sales_extra as se where 1 != 1",
+          "Query": "select se.colb, se.cola, se.`start`, se.`end` from sales_extra as se",
+          "Table": "sales_extra"
+        },
+        {
+          "OperatorType": "Route",
+          "Variant": "Between",
+          "Keyspace": {
+            "Name": "user",
+            "Sharded": true
+          },
+          "FieldQuery": "select s.oid, s.col1 from sales as s where 1 != 1",
+          "Query": "select s.oid, s.col1 from sales as s where s.oid between :se_start /* INT16 */ and :se_end /* INT16 */ and s.col1 = :se_cola /* VARCHAR */",
+          "Table": "sales",
+          "Values": [
+            "(:se_start, :se_end)"
+          ],
+          "Vindex": "binary"
+        }
+      ]
+    },
+    "TablesUsed": [
+      "user.sales",
+      "user.sales_extra"
+    ]
+  },
+  "skip_e2e":true
+}
```

Comment:
new test should not be skipped

Reply:
thanks , out of the [5 new test cases](https://github.com/c-r-dev/vitess/blob/b99ebabe960b49792dd426509d68fbad73a3fda3/go/vt/vtgate/planbuilder/testdata/filter_cases.json#L4972-L5113)  added for `between` , marked the final one related to [join](https://github.com/c-r-dev/vitess/blob/b99ebabe960b49792dd426509d68fbad73a3fda3/go/vt/vtgate/planbuilder/testdata/filter_cases.json#L5064)  as skip_e2e.

Reason for marking this skip , is because of ["OperatorType": "Join"](https://github.com/c-r-dev/vitess/blob/b99ebabe960b49792dd426509d68fbad73a3fda3/go/vt/vtgate/planbuilder/testdata/filter_cases.json#L5070) , verifyTestExpectations for `join` cases verifies -> [Verify that the Join primitive sees atleast 1 row on the left side](https://github.com/c-r-dev/vitess/blob/b99ebabe960b49792dd426509d68fbad73a3fda3/go/test/endtoend/vtgate/plan_tests/main_test.go#L154-L155). Currently we don't have any sample data in these test schema tables. (pls let me know if we can put some test data in these tables for e2e testing)

@harshit-gangal please help to let me know if I have missed any thing , or if there is a better way to write this test for "Between clause on a binary vindex field with values from a different table"

Reply:
We need to add the sample data to have data on left and right side of the join.
The policy after adding these plan test is that we need to ensure that new tests does not add `skip_e2e` flag.
For that we have to keep modifying the sample input data.

Reply:
thanks @harshit-gangal , added sample data loading with https://github.com/vitessio/vitess/pull/17342/commits/1400fea144a1542314e7c1dee4ad1d7798f63ee2 , and removed `skip_e2e`.  All new tests are passing with out skipping.